tests/test_gui.py

```diff
@@ -0,0 +1,1152 @@
+    assert (
+        app_dialog.root.tables.rows["1"]
+        .cells["https://sg-media-staging-usor-01.s3-accelerate.amazonaws.com*"]
+        .exists()
+    ), "Big Buck Bunny project doesn't have a thumbnail"
+    assert (
+        app_dialog.root.tables.rows["1"].cells["False"].exists()
+    ), "Missing Big Buck Bunny project Favorite column"
+    assert (
+        app_dialog.root.tables.rows["1"]
+        .cells[
+            "Big Buck Bunny is a short computer animated film by the Blender Institute, part of the Blender Foundation."
+        ]
+        .exists()
+    ), "Missing Big Buck Bunny project description"
+
+
+def test_entity_field_menu(app_dialog):
+    # Click on the Entity Field Menu widget
+    app_dialog.open_demo_pane("Entity Field Menu")
+    assert app_dialog.root.captions[
+        "Entity Field Menu"
+    ].exists(), "Not on the Entity Field Menu widget"
+
+    # Wait until widget is showing up
+    app_dialog.root.captions["Click the button to show the menu."].waitExist(), 30
+
+    # Validate entity field menu
+    app_dialog.root.buttons["EntityFieldMenu (HumanUser)"].get().mouseClick()
+    time.sleep(5)  # to give some time for the menu to load
+    topwindows.menuitems["Id"].get().mouseClick()
+    app_dialog.root.buttons["EntityFieldMenu (HumanUser)"].get().mouseClick()
+    time.sleep(2)  # to give some time for the menu to load
+    assert topwindows.menuitems["Id"].exists(), "Id not on the entity field menu widget"
+    topwindows.menuitems["Pipeline Configurations"].get().mouseClick()
+
+
+def test_shotgun_menu(app_dialog):
+    # Click on the Shotgun Menu widget
+    app_dialog.open_demo_pane("Shotgun Menu")
+    assert app_dialog.root.captions[
+        "Shotgun Menu"
+    ].exists(), "Not on the Shotgun Menu widget"
+
+    # Wait until widget is showing up
+    app_dialog.root.captions["Click the button to show the menu."].waitExist(), 30
+
+    # Validate Shotgun menu
+    app_dialog.root.buttons["ShotgunMenu"].get().mouseClick()
+    time.sleep(1)  # to give some time for the menu to load
+    assert topwindows.menuitems[
+        "Action 1"
+    ].exists(), "Action 1 not on the Shotgun menu widget"
+    topwindows.menuitems["Submenu"].mouseSlide()
+    assert topwindows.menuitems[
+        "Action 3"
+    ].exists(), "Action 3 not on the Shotgun menu widget"
+    topwindows.menuitems["Action 4"].mouseClick()  # to close the menu
+
+
+def test_spinner(app_dialog):
+    # Click on the Spinner widget
+    app_dialog.open_demo_pane("Spinner")
+    assert app_dialog.root.captions["Spinner"].exists(), "Not on the Spinner widget"
+
+    # Show the spinner
+    app_dialog.root.buttons["spinner.show()"].get().mouseClick()
+    time.sleep(2)  # let the spinner show up
+    # FIXME Need to find a way to know that spinner is showing up
+
+    # Hide the spinner
+    app_dialog.root.buttons["spinner.hide()"].get().mouseClick()
+    # FIXME Need to find a way to know that spinner is hidden
```

Comment:
You should contact Francois Drapeau. He might have some ideas involving the screenshot method.